File: src/ai_anime/modules/platform_release/application/ports.py
"""Platform release outbound ports."""

from __future__ import annotations

from typing import Protocol

from ai_anime.modules.platform_release.domain import ReleaseFeed, ReleaseLocale


class ReleaseFeedPort(Protocol):
    async def current(self, *, locale: ReleaseLocale) -> ReleaseFeed: ...

File: frontend/src/modules/platform_release/application/ports.ts
import type {
  ReleaseFeed,
  ReleaseLocale,
  ReleaseUpgradeStatus,
} from "@/modules/platform_release/domain/release-notifications";

export interface ReleaseNotificationGateway {
  fetch(locale: ReleaseLocale, signal?: AbortSignal): Promise<ReleaseFeed>;
}

export interface ReleaseNotificationStorage {
  isMuted(): boolean;
  markCurrentReleaseSeen(tag: string | null | undefined): void;
  markUpgradeSeen(tag: string | null | undefined): void;
  markUpgradeSkipped(tag: string | null | undefined): void;
  isCurrentReleaseSeen(tag: string | null | undefined): boolean;
  upgradeStatus(tag: string | null | undefined): ReleaseUpgradeStatus;
}
